### DISCUSSION TITLE
Does having hotkeys take an async function make sense?

### DISCUSSION THREAD

**Discussion by trajano** [spyoungtech/ahk#298] [Q&A] [answered]
New to asyncio stuff, but does having `ahk.add_hotkeys` take an async function make sense?  Right now it does not seem to work "quietly".

**Comment by spyoungtech** [comment, marked as answer]
The callback is simply a callable. Although an async function is technically a callable, calling an async function merely produces a coroutine which must be awaited, so this is likely not what you want to do.

If you _must_ call an async function for your callback, you can use different strategies to work around this, (for example, calling `asyncio.run` in a regular function) but ultimately, you want to be passing a regular function to `ahk.add_hotkey`.

Without knowing exactly what your code looks like, a simple solution might be to do something like this:

```
async def my_async_function():
    print('My async function starting')
    await asyncio.sleep(1)
    print('My async function done')

def my_callback():
    asyncio.run(my_async_function())

ahk.add_hotkey('#n', callback=my_callback)
# ...
```

Keep in mind that hotkey callbacks each run in their own thread, anyhow.
